sdk/azcore/runtime/request.go

```diff
@@ -148,6 +153,38 @@ func SetMultipartFormData(req *policy.Request, formData map[string]any) error {
 		return nil
 	}
 
+	writeMultipartContent := func(fieldname, filename string, mpc streaming.MultipartContent) error {
+		if mpc.Body == nil {
+			return errors.New("streaming.MultipartContent.Body cannot be nil")
+		}
+		if mpc.ContentType == "" && mpc.Filename == "" {
+			return writeContent(fieldname, filename, mpc.Body)
+		}
+		if mpc.Filename != "" {
```

Comment:
Why do we have a named parameter named 'filename' and _also_ an optional 'mpc.Filename'? Shouldn't it be one or the other?

Reply:
[`multipart.Writer.CreateFormFile`](https://pkg.go.dev/mime/multipart#Writer.CreateFormFile) requires a file name. We default the file name to the field name. However, tsp has introduced a requirement that apps be able to specify the file name in case the endpoint cares about it.

I should point out that we will only generate `MultipartContent` for tsp. OpenAPI will continue to use `io.ReadSeekCloser` which doesn't allow specifying the file name or Content-Type. We _could_ change it if required (or if we think we should), but it would be a breaking change (I don't know that any real services would be impacted).

Reply:
This is probably just me being dense and not understanding this feature but I also don't get why there are two ways for a caller to specify `filename`. Why would a caller need another argument to override the value it passed as `mpc.Filename`?

Reply:
I was the one being dense, for some reason what Richard pointed out went over my head. I've removed the redundant param.